Atomic/syscontainers.py

```diff
@@ -2779,6 +2785,89 @@ def _correct_dir_permissions_for_user(path):
                     s = os.stat(fullpath)
                     os.chmod(fullpath, s.st_mode | 0o600)
 
+    def _configure_userns(self, config):
+        """
+        Configure the userns for the specified OCI configuration.
+        """
+        config['linux']['namespaces'] = [i for i in config['linux']['namespaces'] if i['type'] != 'network']
+
+        namespaces = set([i['type'] for i in config['linux']['namespaces']])
+
+        # force the 'user' namespace
+        if 'user' not in namespaces:
+            config['linux']['namespaces'].append({'type' : 'user'})
+            namespaces.add('user')
+
+        all_namespaces = set(['pid', 'uts', 'network', 'mount', 'ipc', 'user'])
+
+        drop_sys = len(all_namespaces - namespaces) > 0
+
+        new_mounts = []
+        for i in config['mounts']:
+            if drop_sys and i['destination'].startswith("/sys"):
+                continue
+            if 'options' not in i:
+                i['options'] = []
+
+            if not i['destination'].startswith('/dev'):
```

Comment:
How about `/proc` mount? the default rootless sepc seems not having options like that.
```
{
                        "destination": "/proc",
                        "type": "proc",
                        "source": "proc"
                },
```

Reply:
for `/proc` it doesn't make any difference, as there are no devices under `/proc`.